src/App.php

```diff
@@ -311,12 +318,14 @@ function ( $wikiLang ) use ( $slim ) {
 					$slim->authManager->logout();
 					$slim->redirect( $slim->urlFor( 'root' ) );
 				} )->name( 'logout' );
-				$slim->get( ':wikiLang/loadmore', function ( $wikiLang ) use ( $slim ) {
-					$page = new Controllers\CopyPatrol( $slim );
-					$page->setDao( $this->getPlagiabotDao() );
-					$page->setWikiDao( $this->getWikiDao( $wikiLang ) );
-					$page();
-				} )->name( 'loadmore' )->setConditions( $routeConditions );
+				$slim->get( ':wikiLang/loadmore', $middleware['valid-wikiLang'],
+					function ( $wikiLang ) use ( $slim ) {
+						$page = new Controllers\CopyPatrol( $slim );
+						$page->setDao( $this->getPlagiabotDao() );
+						$page->setWikiDao( $this->getWikiDao( $wikiLang ) );
+						$page();
+					}
+				)->name( 'loadmore' )->setConditions( $routeConditions );
```

Comment:
I trust your judgement! The extra group does mean another layer of nesting, which could be rough on the eyes. Up to you